src/transforms/dedupe.rs

```diff
@@ -103,7 +103,10 @@ fn default_cache_config() -> CacheConfig {
 //   structure can vary significantly. This should probably either become a required field
 //   in the future, or maybe the "semantic meaning" can be utilized here.
 fn default_match_fields() -> Vec<String> {
-    let mut fields = vec![log_schema().message_key().into()];
+    let mut fields = vec![];
```

Comment:
Prefer avoiding the overhead of the macro for empty arrays:
```suggestion
    let mut fields = Vec::new();
```